modules/internal/ChapelArray.chpl

```diff
@@ -2712,6 +2721,13 @@ module ChapelArray {
       }
     }
 
+    pragma "no doc"
+    /* Ensure promoted-push_front is serial */
+    proc push_front(vals) where isIterable(vals) {
+      for val in vals do
```

Comment:
Yeah, that's tricky... If users think about this like a "promoted" `push_front`, then they might expect the current implementation (`for val in vals`). If they think about this like a way to prepend to an array, then they might expect the order to be preserved.

The latter seems more useful, while the former seems less surprising IMO.

Reply:
My gut reaction is that we should preserve the order of elements (i.e. the first element yielded is the first element of the array after push_front). While this is simple for arrays, the implementation will be more difficult for iterators yielding an unknown number of elements.

Do we want to split array push_front/back to a separate method with unique behavior, and have the isIterable method as a way to avoid the need for locking in promoted expressions?